Google.Api.Gax.Grpc/GrpcAdapter.cs

```diff
@@ -38,5 +43,27 @@ public ChannelBase CreateChannel(string endpoint, ChannelCredentials credentials
         /// <param name="options">The channel options to use. Will not be null.</param>
         /// <returns>A channel for the specified settings.</returns>
         protected abstract ChannelBase CreateChannelImpl(string endpoint, ChannelCredentials credentials, GrpcChannelOptions options);
+
+        /// <summary>
+        /// Returns the default gRPC adapter based on the available gRPC implementations.
+        /// </summary>
+        public static GrpcAdapter DefaultAdapter => s_defaultFactory.Value;
+
+        // TODO: Is this really what we want? Definitely simple, but not great in other ways...
+        // Perhaps have an environment variable that can be used to force one or the other?
+        // (In some situations it may be better to see "Grpc.Core fails immediately, but that's the one we want"
+        // rather than "Grpc.Net.Client looks like it will work, but then fails".)
+        private static GrpcAdapter CreateDefaultAdapter()
+        {
+            try
+            {
+                GrpcChannel.ForAddress("https://ignored.com");
```

Comment:
If this succeeds, can we guarantee that Grpc.Net.Client is fully usable?

Reply:
That's something we need to check with Jan. But I'm pretty sure we'll want some what of forcing it one way or another globally - an environment variable being the simplest approach.